contracts/staking/TokenStaking.sol

```diff
@@ -0,0 +1,1427 @@
+        (, uint256 createdAt, ) = keepStakingContract.getDelegationInfo(
+            operator
+        );
+        require(createdAt != 0, "Nothing to sync");
+
+        uint256 keepStakeAmount = keepStakingContract.eligibleStake(
+            operator,
+            address(this)
+        );
+        uint96 tAmount = 0;
+        if (keepStakeAmount > 0) {
+            (tAmount, ) = keepToT(keepStakeAmount);
+        }
+        return tAmount;
+    }
+
+    /// @notice Returns the T token amount that's obtained from `amount` wrapped
+    ///         tokens (KEEP), and the remainder that can't be converted.
+    function keepToT(uint256 keepAmount)
+        internal
+        view
+        returns (uint96 tAmount, uint256 keepRemainder)
+    {
+        keepRemainder = keepAmount % keepFloatingPointDivisor;
+        uint256 convertibleAmount = keepAmount - keepRemainder;
+        tAmount = ((convertibleAmount * keepRatio) / keepFloatingPointDivisor)
+            .toUint96();
+    }
+
+    /// @notice The amount of wrapped tokens (KEEP) that's obtained from
+    ///         `amount` T tokens, and the remainder that can't be converted.
+    function tToKeep(uint96 tAmount)
+        internal
+        view
+        returns (uint256 keepAmount, uint96 tRemainder)
+    {
+        tRemainder = (tAmount % keepRatio).toUint96();
+        uint256 convertibleAmount = tAmount - tRemainder;
+        keepAmount = (convertibleAmount * keepFloatingPointDivisor) / keepRatio;
+    }
+
+    /// @notice Returns the T token amount that's obtained from `amount` wrapped
+    ///         tokens (NU), and the remainder that can't be converted.
+    function nuToT(uint256 nuAmount)
+        internal
+        view
+        returns (uint96 tAmount, uint256 nuRemainder)
+    {
+        nuRemainder = nuAmount % nucypherFloatingPointDivisor;
+        uint256 convertibleAmount = nuAmount - nuRemainder;
+        tAmount = ((convertibleAmount * nucypherRatio) /
+            nucypherFloatingPointDivisor).toUint96();
+    }
+
+    /// @notice The amount of wrapped tokens (NU) that's obtained from
+    ///         `amount` T tokens, and the remainder that can't be converted.
+    function tToNu(uint96 tAmount)
+        internal
+        view
+        returns (uint256 nuAmount, uint96 tRemainder)
+    {
+        //slither-disable-next-line weak-prng
+        tRemainder = (tAmount % nucypherRatio).toUint96();
+        uint256 convertibleAmount = tAmount - tRemainder;
+        nuAmount =
+            (convertibleAmount * nucypherFloatingPointDivisor) /
+            nucypherRatio;
+    }
+
+    /// @notice Extract owner of Keep stake. Possible ways: grantee in ManagedGrant,
+    ///         grantee in TokenGrant or owner in Keep staking contract
+    /// @return keepOwner Owner of stake
+    /// @return isManagedGrant Returns true if owner is grantee in ManagedGrant contract
+    function getKeepOwner(address operator)
```

Comment:
See https://github.com/threshold-network/solidity-contracts/pull/26 please. I think we should replace the content of this function with a call to that library and leave the rest as-is.

I could not make a PR to this branch because it's already a fork, so I suggest merging #26 first and then merging `main` changes here.

Reply:
✔️